src/Initializer/ParameterDB.cpp

```diff
@@ -250,11 +331,121 @@ namespace seissol {
     void MaterialParameterDB<T>::evaluateModel(std::string const& fileName, QueryGenerator const& queryGen) {
       easi::Component* model = loadEasiModel(fileName);
       easi::Query query = queryGen.generate();
-
+      
       easi::ArrayOfStructsAdapter<T> adapter(m_materials->data());
       addBindingPoints(adapter);
-      model->evaluate(query, adapter); 
-      
+      model->evaluate(query, adapter);
+
+      delete model;
+    }
+
+    template<>
+    void MaterialParameterDB<seissol::model::ElasticMaterial>::evaluateModel(std::string const& fileName, QueryGenerator const& queryGen) {
+      easi::Component* model = loadEasiModel(fileName);
+      easi::Query query = queryGen.generate();
+      const unsigned numPoints = query.numPoints();
+
+      std::vector<seissol::model::ElasticMaterial> elasticMaterials(numPoints);
+      easi::ArrayOfStructsAdapter<seissol::model::ElasticMaterial> adapter(elasticMaterials.data());
+      MaterialParameterDB<seissol::model::ElasticMaterial>().addBindingPoints(adapter);
+      model->evaluate(query, adapter);
+
+      // Only use homogenization when ElementAverageGenerator has been supplied
+      if (const ElementAverageGenerator* gen = dynamic_cast<const ElementAverageGenerator*>(&queryGen)) {
+        const unsigned numElems = numPoints / NUM_QUADPOINTS;
+        std::array<double, NUM_QUADPOINTS> quadratureWeights{ gen->getQuadratureWeights() };
+        std::vector<double> elemVolumes{ gen->getElemVolumes() };
+        std::array<double, 3> matMeanInit{ 0, 0, 0 };
+        std::vector<seissol::model::ElasticMaterial> materialsMean(numElems, seissol::model::ElasticMaterial{ matMeanInit.data(), 3 });
+        std::vector<double> vERatioMean(numElems);
+
+        // Approximate element volume integrals using Gaussian quadrature
+        for (unsigned i = 0; i < numPoints; ++i) {
+          // Scale up quadrature weights by (element volume / reference volume)
```

Comment:
I would replace the comment with a (const) variable quadratureScale = element volumne / reference volume